exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -42,7 +43,19 @@ type exporter struct {
 	retryCount       int
 	collectorID      string
 	svcStructuredLog *cwlogs.Client
-	pusher           cwlogs.Pusher
+	pusherMap        map[cwlogs.PusherKey]cwlogs.Pusher
+	pusherMapLock    sync.Mutex
```

Comment:
I'm uncertain if the mutex is needed. This could be a little cargo cult programming. Does consume logs run multi treaded?

Reply:
It is possible for `ConsumeLogs()` to be invoked multiple times concurrently, yes.